src/astro/files/base.py

```diff
@@ -142,15 +144,21 @@ def resolve_file_path_pattern(
     :param normalize_config: parameters in dict format of pandas json_normalize() function
     """
     location = create_file_location(path_pattern, conn_id)
-    files = [
-        File(
-            path=path,
-            conn_id=conn_id,
-            filetype=filetype,
-            normalize_config=normalize_config,
+
+    files = []
+    for path in location.paths:
+        # Skip file object with folder paths
+        if path.endswith("/"):
+            continue
+
+        files.append(
+            File(
+                path=path,
+                conn_id=conn_id,
+                filetype=filetype,
+                normalize_config=normalize_config,
+            )
```

Comment:
```suggestion

    files = [
        File(
            path=path,
            conn_id=conn_id,
            filetype=filetype,
            normalize_config=normalize_config,
        )
        for path in location.paths if not path.endswith("/")
    ]
```

Reply:
Logically same but uses list comprehension

Reply:
Ohh, I also added the same change there: https://github.com/astronomer/astro-sdk/pull/740/commits/124c42f624b6d7867a3e37c94a81335b04a67d5a. Sorry for that.